### PR TITLE
PUD-969: Use user_uuid from auth token rather than requiring client to provide it

### PR DESCRIPTION
**Pull request by philsegal**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/RecallsController.kt [hunk 1]
```diff
@@ -48,15 +55,28 @@ class RecallsController(
...
+  fun bookRecall(@RequestBody bookRecallRequest: BookRecallRequest, @RequestHeader("Authorization") bearerToken: String): ResponseEntity<RecallResponse> {
+    val token = getTokenFromHeader(bearerToken)
+
+    return ResponseEntity(
+      recallRepository.save(bookRecallRequest.toRecall(token.userUuid)).toResponse(),
       HttpStatus.CREATED
     )
+  }
+
+  private fun getTokenFromHeader(bearerToken: String): Token {
```
Comment (AndyNewton) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/RecallsController.kt]: "Obvs" move this to a helper/the bean/don't copy to each controller - as a future refactoring.

Comment (AndyNewton) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/RecallsController.kt]: Also, I guess the "real" test of this is the E2Es.

**Review by AndyNewton — APPROVED** [review]
Soz for delay.  Looks OK, albeit a bit 'black magic'.  Couple of minor comments.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/RecallsController.kt (hunk 1 above)
Comment (philsegal) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/RecallsController.kt]: Yep,  have done so in the next commit :)

**Pull request merged**